### PR TITLE
[3] fix(2892): API tokens are correctly authenticated with new sd admins format

### PR DESCRIPTION
**Pull request by y-oksaku**
## Context

The new sd admins format was introduced by this PR.
https://github.com/screwdriver-cd/screwdriver/pull/2854

But it cannot authenticate API token has admin permission.

## Objective

Fix to authenticate correctly even when using API tokens.

## References

Please review the following PRs first.

- https://github.com/screwdriver-cd/data-schema/pull/518
- https://github.com/screwdriver-cd/scm-github/pull/216

Issue

- https://github.com/screwdriver-cd/screwdriver/issues/2892

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/auth/index.js [hunk 1]
```diff
@@ -204,10 +205,26 @@ const authPlugin = {
...
                         }
 
+                        let scmUser = {};
+
+                        try {
+                            scmUser = await scm.decorateAuthor({
+                                username: user.username,
+                                scmContext: user.scmContext,
+                                token: await user.unsealToken()
+                            });
```
Comment (y-oksaku) [inline plugins/auth/index.js:215]: Retrieving the latest user id so that it cannot be impersonated.

File: plugins/auth/index.js [hunk 2]
```diff
@@ -204,10 +205,26 @@ const authPlugin = {
...
+                                ['auth', 'error'],
+                                `Fails to find the user "${user.username}" in ${user.scmContext}.`
+                            );
+                        }
+
                         await createDefaultCollection(user, collectionFactory);
 
                         profile = {
                             username: user.username,
+                            scmUserId: scmUser.id,
```
Comment (y-oksaku) [inline plugins/auth/index.js:227]: Include the `scmUserId` in the profile same as `scmContext` to get `scmUserId` when authenticating with this JWT.

**Review by ibu1224 — APPROVED** [review]

**Review by itleigns — APPROVED** [review]

File: plugins/auth/index.js [hunk 3]
```diff
@@ -204,10 +205,26 @@ const authPlugin = {
...
 
+                        let scmUser = {};
+
+                        try {
+                            scmUser = await scm.decorateAuthor({
+                                username: user.username,
+                                scmContext: user.scmContext,
+                                token: await user.unsealToken()
+                            });
+                        } catch (e) {
```
Comment (klu909) [inline plugins/auth/index.js:216]: shouldn't we return false here?

Comment (y-oksaku) [reply plugins/auth/index.js]: Thank you. I fixed.

**Review by tkyi — APPROVED** [review]

**Review by kumada626 — APPROVED** [review]

**Pull request merged**